lib/util/memlimit.c

```diff
@@ -216,8 +216,11 @@ memlimit_sysconf(size_t * memlimit)
 	/* Read the two limits. */
 	if (((pagesize = sysconf(_SC_PAGE_SIZE)) == -1) ||
 	    ((physpages = sysconf(_SC_PHYS_PAGES)) == -1)) {
-		/* Did an error occur? */
-		if (errno != 0)
+		/*
+		 * Did an error occur?  OS X may return EINVAL due to not
```

Comment:
I considered that, but if other OS kernels return EINVAL I'd want to handle it the same way.  The bug here is in the header files (they shouldn't define _SC_ macros which the kernel doesn't support) rather than in the kernel.